lib/src/header_parser/sub_parsers/compounddecl_parser.dart

```diff
@@ -326,6 +339,19 @@ int _compoundMembersVisitor(clang_types.CXCursor cursor,
   return clang_types.CXChildVisitResult.CXChildVisit_Continue;
 }
 
+final _paramNameList = <String>[];
+
+int _functionPointerFieldVisitor(clang_types.CXCursor cursor,
+    clang_types.CXCursor parent, Pointer<Void> clientData) {
+  if (cursor.kind == clang_types.CXCursorKind.CXCursor_ParmDecl) {
+    final spelling = cursor.spelling();
+    if (spelling.isNotEmpty) {
+      _paramNameList.add(spelling);
+    }
+  }
+  return clang_types.CXChildVisitResult.CXChildVisit_Continue;
+}
+
```

Comment:
> Then it would be a one-liner to add parameter names to function pointers in function arguments as well in functiondecl_parser.

I am only cursorily (!) familiar with ffigen codebase. Can you elaborate more?

Reply:
If we separate this out to another sub_parser, we can call that function here as well
https://github.com/dart-lang/ffigen/blob/7f14959e8cfe799fd31524ce6b9b36256cad8e0b/lib/src/header_parser/sub_parsers/functiondecl_parser.dart#L102-L107

Something like this -
```dart
if (pt is PointerType && pt.child is NativeFunc) {
    parseParametersFromFunctionPointerCursor(pt, paramCursor);
}
```

Reply:
On a side note, should I

```dart
if (pt.baseType is NativeFunc) {
  /// ... 
}
```

because somewhere double pointer to function is used?

Reply:
Yeah we should probably handle pointer relations, and there could be multiple levels of Typedefs too.

Reply:
> I think it would make more sense to add a variation of getCodegenType(CXCursor) than thi

Yeah you can do that too.

Well I guess using a stack doesn't seem to be necessary in this case (since parsing the types and adding the parameters is a different step) but I only suggested it since other sub parsers follow a similar structure.

Reply:
> Well I guess using a stack doesn't seem to be necessary

If we came this far, why not handle nested function pointers?

I am in favor of changing `toCodeGenType` even if it takes longer. Because we would then be able to have param names in function pointer typedefs, function pointers as parameters etc..

Reply:
Yeah that would actually be a complete solution.